src/port1.0/portconfigure.tcl

```diff
@@ -715,6 +1084,47 @@ proc portconfigure::configure_get_compiler {type {compiler {}}} {
             objcxx  { return ${prefix}/bin/llvm-g++-4.2 }
             cpp     { return ${prefix}/bin/llvm-cpp-4.2 }
         }
+    } elseif {[regexp {^macports-g95$} $compiler]} {
+        switch $type {
+            fc      -
+            f77     -
+            f90     { return ${prefix}/bin/g95 }
+        }
+    } elseif {[regexp {^macports-([^-]+)-clang$} $compiler -> mpi]} {
```

Comment:
Maybe use `^macports-(mpich|openmpi)-clang$` to make it clear what is expected here? At least that is what I assume is meant to be matched. Same for the following conditions.

Reply:
You are quite correct.
Regular expressions are difficult enough.
It should be fixed in 21464ee0bd26ecb5a2acd6333883bfc9a2028456.